library/spdm_transport_storage_lib/libspdm_storage.c

```diff
@@ -0,0 +1,819 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+/**
+ *  Author: Wilfred Mallawa <wilfred.mallawa@wdc.com>
+ *          Alistair Francis <alistair.francis@wdc.com>
+ **/
+
+#include "library/spdm_transport_storage_lib.h"
```

Comment:
Do you have any unit test associated to prove the correctness of this function?

Reply:
I haven't written any explicit test cases for all the functions in this files. We have tested the logic by testing with QEMU + `spdm-utils` + linux (Kernel SPDM) though. I'm waiting for the spec to become ratified so we can add some of the changes discussed above.

I can add some tests if they are required? Are there exists tests for other transports?

Reply:
Usually, we will need two level of test:
1) Unit test in libspdm project.
2) System test in spdm-emu project.

In history, both tests can catch some error.

Please consider to add both.